scripts/scDblFinder.Rmd

```diff
@@ -0,0 +1,204 @@
+---
+title: "ScDblFinder filtering with recoverDoublets - `r params$sample_name`"
+author: 'Antonia Chroni <chronia@chop.edu> for D3B'
+date: "`r Sys.Date()`"
+output:
+  html_notebook:
+    toc: TRUE
+    toc_float: TRUE
+params:
+  data_path: './' # path to the directory with the `seurat_obj_raw.rds` file from the `seurat_alignment_qc.Rmd` script
+  results_dir: './' # path to dir to save all output files
+  sample_name: "test"   # name of the sample in the library
+  log_file: NULL
+---
+
+# Information about this notebook
+This script used the [ScDblFinder](https://bioconductor.org/packages/devel/bioc/vignettes/scDblFinder/inst/doc/scDblFinder.html) method for identifying doublets in single-cell data. 
+
+# Usage
+To run the Rscript from the command line sequentially, use:
+
+```
+Rscript -e "rmarkdown::render('scDblFinder.Rmd', clean = TRUE,
+      params=list(results_dir = '/Users/chronia/CHOP/GitHub/kf-dev-single-cell-rna/scripts/example/', 
+                  data_path = '/Users/chronia/CHOP/GitHub/kf-dev-single-cell-rna/scripts/example/test.seurat_obj_raw.rds',
+                  sample_name = 'test'))"
+```
+
+# Set up
+
+```{r load-library, echo=TRUE}
+suppressPackageStartupMessages({
+  library(scDblFinder)
+  library(Seurat)
+  library(scater)
+  library(future)
+  library(tidyverse)
+  library(grid)
+})
+
+# # evaluate Seurat R expressions asynchronously when possible (such as ScaleData) using future package
+plan("multisession", workers = 4)
+options(future.globals.maxSize = 64000 * 1024^2) # for 64 Gb RAM
```

Comment:
These should probably be tied to the actual CPU and RAM values being passed to the software.